controllers/providers/aws/aws.go

```diff
@@ -238,3 +241,12 @@ func GetInstanceTypeNetworkInfo(instanceTypes []*ec2.InstanceTypeInfo, instanceT
 	}
 	return nil
 }
+
+func GetInstanceTypeArchetectures(instanceTypes []*ec2.InstanceTypeInfo, instanceType string) []*string {
+	for _, instanceTypeInfo := range instanceTypes {
```

Comment:
probably makes sense to move the instanceType lookup into a helper function since we're doing the same thing in the function above.